### DISCUSSION TITLE
Welcome to BootstrapBlazor Discussions!

### DISCUSSION THREAD

**Discussion by ArgoZhang** [dotnetcore/BootstrapBlazor#283] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ArgoZhang** [comment]
大家加油，把这块地方利用起来

**Reply by densen2014** [reply]
- [x] do it

**Comment by dmordaque** [comment]
Hi the team, congrat for your work ! 
I would like to use your components but i haven't see the possibility to custom the bootstrap style. 
It's important for my compagny to keep our branding and all custom boostrap code that we have created. There is possibilty to custom ? If not is it palnned ?
I profite of my message for another question, is it planned to have grouping featue for table ?
Thx for your aswer.
Damien.

**Reply by ArgoZhang** [reply]
hi Damien

Thank you for contacting us.

### `BootstrapBlazor` full support for custom the bootstrap style

#### Global
1. via `bootstrap package manager`
link https://getbootstrap.com/docs/5.3/customize/overview/

2. custom style sheet
```
:root {
    --bs-variable: value;
    --bb-variable: value;
}
```

#### Individual

1. custom style sheet in your owner project

`~/site.css`

custom the `Button` component style link the following

```
.btn {
    --bs-btn-font-size: .875rem;
    --bs-btn-focus-box-shadow: none;
    --bb-btn-label-margin-left: 4px;
    --bs-btn-active-border-color: transparent;
    --bs-btn-hover-border-color: transparent;
    --bs-btn-border-radius: var(--bs-border-radius);
}
```

### Grouping for Table

not yet for now. Do you have a good solution or PR for us

**Comment by SalvatoreFonti** [comment]
Hello BootstrapBlazor team.
Thank you and congratulations on this product.

I'm currently implementing a project based on Blazer Server Site.
Your BootstrapBlazor.BarcodeReader component is used there and it works
wonderful. On Android devices it is a bit annoying that you always have to select the backcamera
from Dropdown Field.Is there a way to initialise the backcamera on page load, 
so that you can scan directly without having to change the camera?
Is there a way to get the camera device id/name from selected camera device?

Thanks very much!
Salva

**Reply by ArgoZhang** [reply]
@SalvatoreFonti hi. I have an idea for this. let me try. later reply.

**Comment by ArgoZhang** [comment]
@SalvatoreFonti 

hi. i have checked the `ZXing` lib source code. it's use the back camera by default if not have the exact `DeviceId`.

https://github.com/zxing-js/library/blob/99a8e0c65de7bf97565a5dd46299d858c10dd69a/src/browser/BrowserCodeReader.ts#L332

and I checked the sample on https://www.blazor.zone/barcodereaders#Normal it's open the back camera by default too. (on iOS)

Is that what happens with Android phones? Can you try it on an iPhone?

**Reply by SalvatoreFonti** [reply]
Thank you for the info!

It seems that the first camera from the Devices Enumeration is always initialized.
See attachment, tested on Google Pixel 7 and Samsung S20 with Android 13.

I could not test with IOS.

Unfortunately, I can't tell the BootstrapBlazor.Components.BarcodeReader component a device ID
with which the component should initialize. Otherwise you could first determine the device ID of the back camera
and initialize the BarcodeReader with the ID.

Thanks again!
Salva

![Pixel_7_20230426](https://user-images.githubusercontent.com/37700136/234601765-76954f1b-1e53-426b-ba1e-66bfb1cbdad6.png)
![Samsung_S20_20230426](https://user-images.githubusercontent.com/37700136/234601772-65582451-d7cb-427b-89be-5131d8843715.jpg)

**Reply by ArgoZhang** [reply]
thanks for you provided the screenshot it's very useful! i will add a parameter for you that you can set the first initialized device like `last` or `first`. what do you think?

**Reply by SalvatoreFonti** [reply]
That would be a real relief and makes the component perfect for our purposes.
Many Thanks for your effort!

**Reply by ArgoZhang** [reply]
link #1149
